include/soci/blob.h

```diff
@@ -57,22 +57,22 @@ class SOCI_DECL blob
     // Note: Using an offset > 0 on a blob whose size is less than
     // or equal to offset, will throw an exception.
     std::size_t read_from_start(void * buf, std::size_t toRead,
-        std::size_t offset = 0);
+        std::size_t offset = 0) const;
 
     // offset is backend-specific
 #ifndef SOCI_SOURCE
     [[deprecated("Use write_from_start instead")]]
 #endif
     std::size_t write(std::size_t offset, const void * buf,
-        std::size_t toWrite);
+        std::size_t toWrite) const;
```

Comment:
It seems weird to make `write()` const, even if it doesn't modify the BLOB itself, it still clearly modifies something. Why do we want to do this?